src/compression_with_clause.h

```diff
@@ -19,6 +19,7 @@ typedef enum CompressHypertableOption
 	CompressSegmentBy,
 	CompressOrderBy,
 	CompressChunkTimeInterval,
+	_CompressOptionMax
```

Comment:
sure; I can remove the `_` ; but in this repository there are a lot of enums which has an underscore prefix in their Max; I just followed the existing convention...see [here](https://github.com/timescale/timescaledb/blob/348796f9d9dcfc01716e78886ac1cb3de8623f8f/src/ts_catalog/catalog.h#L61) for example

Reply:
For nits like this, it is your decision if you want to do it or not.

Reply:
I've renamed it...but I find it a bit confusing - without any distinctive things in the name its not clear at first glance that this enum entry supposed to mark the max value and its not there because there is a 'max' in the options....but since its the last and I guess everyone working on this will kinda know what could that be - I don't think thats a big hazard..

I looked at all the existing enum keys which have 'max' in their name; and all of them have a `_`  prefix:
```
git grep  -EC1  '^\s*[A-Za-z_]+M[Aa][xX]'  $(find . -name *.h)
```

however these 2 enums doesn't follow convention by using snake-case in their names anyway; so it's mixed up anyway..maybe all the enum keys should be renamed to follow the existing conventions?